ui/src/pages/version/components/forms/components/transformer/InputPanel.js

```diff
@@ -0,0 +1,142 @@
+import React from "react";
+import {
+  EuiDragDropContext,
+  euiDragDropReorder,
+  EuiDraggable,
+  EuiDroppable,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer
+} from "@elastic/eui";
+import { get, useOnChangeHandler } from "@gojek/mlp-ui";
+import { AddButton } from "./components/AddButton";
+import { TableInputCard } from "./components/table_inputs/TableInputCard";
+import { VariablesInputCard } from "./components/table_inputs/VariablesInputCard";
+import { FeastInputCard } from "../feast_config/components/FeastInputCard";
+import { Panel } from "../Panel";
+import {
+  FeastInput,
+  TablesInput
+} from "../../../../../../services/transformer/TransformerConfig";
+import { FeastResourcesContextProvider } from "../../../../../../providers/feast/FeastResourcesContext";
+
+export const InputPanel = ({ inputs = [], onChangeHandler, errors = {} }) => {
+  const { onChange } = useOnChangeHandler(onChangeHandler);
+
+  const onAddInput = (field, input) => {
+    onChangeHandler([...inputs, { [field]: input }]);
+  };
+
+  const onDeleteInput = idx => () => {
+    inputs.splice(idx, 1);
+    onChangeHandler([...inputs]);
+  };
+
+  const onDragEnd = ({ source, destination }) => {
+    if (source && destination) {
+      const items = euiDragDropReorder(inputs, source.index, destination.index);
+      onChangeHandler([...items]);
+    }
+  };
+
+  return (
+    <Panel title="Input" contentWidth="80%">
+      <EuiSpacer size="xs" />
+
+      <EuiFlexGroup direction="column" gutterSize="s">
+        <EuiDragDropContext onDragEnd={onDragEnd}>
+          <EuiDroppable droppableId="INPUTS_DROPPABLE_AREA" spacing="m">
+            {inputs.map((input, idx) => (
+              <EuiDraggable
+                key={`${idx}`}
+                index={idx}
+                draggableId={`${idx}`}
+                customDragHandle={true}
+                disableInteractiveElementBlocking>
+                {provided => (
+                  <EuiFlexItem key={`input-${idx}`}>
+                    {input.feast && (
+                      <FeastResourcesContextProvider
+                        project={input.feast[0].project}>
+                        <FeastInputCard
+                          index={idx}
+                          table={input.feast[0]}
+                          tableNameEditable={true}
+                          onChangeHandler={onChange(`${idx}.feast.0`)}
+                          onDelete={
+                            inputs.length > 1 ? onDeleteInput(idx) : undefined
+                          }
+                          dragHandleProps={provided.dragHandleProps}
+                          errors={get(errors, `${idx}.feast.0`)}
+                        />
+                      </FeastResourcesContextProvider>
+                    )}
+
+                    {input.tables && (
+                      <TableInputCard
+                        index={idx}
+                        table={input.tables[0]}
```

Comment:
Yes. When the user click Add Generic Table again, it will create another `input` that contains only one `tables`. The config would be like this:

```
transformerConfig:
  preprocess:
    inputs:
      - tables:
          - name: ''
            columns: []
      - tables:
          - name: ''
            columns: []
```